### DISCUSSION TITLE
How do I add a form at the end of each post?

### DISCUSSION THREAD

**Discussion by jarmos-san** [adityatelange/hugo-PaperMod#250] [Q&A] [answered]
I've organized my blog posts under `content/posts` which are distinct & different from the other pages under `content/`. How do I configure Hugo to add a `form` only to the those files generated from under `content/posts`?

I tried adding my own customized `single.html` under `layouts/_default` but it adds the `form` to all the contents under `content/` (except the `archive.md` which is great). Here's the file tree of the `contents` directory.

```
content/
| - posts/
|    | - article-1.md
|    | - ...
|    | - article-n.md
| - about.md
| - archives.md
| - projects.md
| - services.md
| - sponsorship.md
```

All I want is, for Hugo to append the `form` to the contents under `content/posts` i.e `article-1.md` to `article-n.md` leaving the all the other files under `content/` as is after the HTML files are generated.

**Comment by kdkasad** [comment, marked as answer]
#### Option 1
In `single.html`, you could try something like the following:
```go
{{- if (eq .Type "posts") -}}
{{ partial "form.html" . }}
{{/* or include HTML for the form here */}}
{{- end -}}
```

#### Option 2
Or, what I would recommend, is using a custom `param` to handle this:
```go
{{- if .Param "enableForm" -}}
{{ partial "form.html" . }}
{{/* or include HTML for the form here */}}
{{- end -}}
```
Then you'd add the following to each page's front matter:
```yaml
enableForm: true
```

**Reply by jarmos-san** [reply]
Alright, your advices are pretty spot on! But here're some inconsistencies I observed.

- I prefer option 2 as well since it lets me configure the site through configuration variables. But it doesn't work as expected. The form still shows up on the pages, it's not supposed to.
- Option 1 works flawlessly.

So, to add a form only to the blog posts that are under `content/posts` creating a `div` in the `single.html` file with the code from option 1 should be to go-to workaround.

**Reply by kdkasad** [reply]
> * I prefer option 2 as well since it lets me configure the site through configuration variables. But it doesn't work as expected. The form still shows up on the pages, it's not supposed to.

 That's weird. I'm not sure why that happens. But since Option 1 works for you, I don't feel like figuring out why.

**Reply by jarmos-san** [reply]
> But since Option 1 works for you, I don't feel like figuring out why.

True! 😆 

Regardless, cheers for the heads up. Really appreciate all the help.
